dix/cursor.c

```diff
@@ -195,20 +196,21 @@ RealizeCursorAllScreens(CursorPtr pCurs)
                     DeviceIntPtr pDevIt = inputInfo.devices;    /*dev iterator */
 
                     while (pDevIt && pDevIt != pDev) {
-                        if (DevHasCursor(pDevIt))
-                            (*walkScreen->UnrealizeCursor) (pDevIt, walkScreen, pCurs);
+                        if (DevHasCursor(pDevIt) && walkScreen->UnrealizeCursor)
+                            walkScreen->UnrealizeCursor(pDevIt, walkScreen, pCurs);
                         pDevIt = pDevIt->next;
                     }
                     while (--nscr >= 0) {
                         walkScreen = screenInfo.screens[nscr];
                         /* now unrealize all devices on previous screens */
                         pDevIt = inputInfo.devices;
                         while (pDevIt) {
-                            if (DevHasCursor(pDevIt))
-                                (*walkScreen->UnrealizeCursor) (pDevIt, walkScreen, pCurs);
+                            if (DevHasCursor(pDevIt) && walkScreen->UnrealizeCursor)
+                                walkScreen->UnrealizeCursor(pDevIt, walkScreen, pCurs);
                             pDevIt = pDevIt->next;
                         }
-                        (*walkScreen->UnrealizeCursor) (pDev, walkScreen, pCurs);
+                        if (walkScreen->UnrealizeCursor)
+                            walkScreen->UnrealizeCursor(pDev, walkScreen, pCurs);
```

Comment:
Alternatively we could do `if (!walkScreen->UnrealizeCursor) continue;` at the start of the loop (line 204-205), if I don’t miss anything.